### PR TITLE
Update `version.txt` on master branch during release workflow

### LINKED ISSUE #3998: Updated version of `setuptools` rejects the `version.txt` ("dev") from `master`
The following error is currently affecting open PRs:

```
Installing spinalcordtoolbox...

Obtaining file:///home/runner/work/spinalcordtoolbox/spinalcordtoolbox
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running egg_info
      /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/dist.py:543: UserWarning: The version specified ('dev') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
        warnings.warn(
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/setup.py", line 16, in <module>
          setup(
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/dist.py", line 1208, in run_command
          super().run_command(command)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.ensure_finalized()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 111, in ensure_finalized
          self.finalize_options()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 219, in finalize_options
          parsed_version = parse_version(self.egg_version)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/version.py", line 266, in __init__
          raise InvalidVersion(f"Invalid version: '{version}'")
      pkg_resources.extern.packaging.version.InvalidVersion: Invalid version: 'dev'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Failed running pip install: 1


Installation failed!

Please find the file "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/install_sct_log.txt",
then upload it as a .txt attachment in a new topic on SCT's forum:
--> http://forum.spinalcordmri.org/c/sct

Error: Process completed with exit code 1.
```

Our `version.txt` on master is the following:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/1b5c49c3904dffe63546adbadc47759a53fbc8ca/spinalcordtoolbox/version.txt#L1

Investigation to come -- just quickly opening up an issue.

### PR DESCRIPTION
**Pull request by joshuacwnewton**
## Description

This PR is a follow-up to https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3999. It makes sure that our `version.txt` on `master` is updated when a new release is created, to reflect the version currently in development.

## Linked issues

Fixes #3998.

### CODE REVIEW

**Comment by joshuacwnewton** [comment]
I'm going to switch this PR to draft while I take a look at the suggestion of `setuptools-scm` from https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3998#issuecomment-1387444957.

**Comment by joshuacwnewton** [comment]
I've taken a look at `setuptools-scm`, and I'm not sure that it's suited for our purposes: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3998#issuecomment-1409026577

So, I'm marking this PR as ready-for-review once more.

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -128,29 +131,34 @@ jobs:
...
       uses: actions/checkout@v2
       with:
         ref: ${{ env.MAIN_BRANCH }}
         fetch-depth: 0
 
-    - name: Download requirements-freeze.txt
+    - name: Update version.txt (main branch)
+      run: | 
+          echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
+          git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
```
Comment (mguaypaq) [inline .github/workflows/create-release.yml:149]: Missing `git add`:
```suggestion
          echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
          git add spinalcordtoolbox/version.txt
          git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
```

**Review by mguaypaq — COMMENTED** [review]
Would you mind if I changed the steps a little bit? I'll add an extra commit to this branch as a proof of concept, and you can tell me whether you like it or not.

My reason is, with the current proposed set of steps, I think the commit history would look like this:
```
                                       "5.8" (release)
                                     /
                                    /
"5.8.dev0" -- <stuff> -- "5.9.dev0" -- <stuff> (main)
```
But I would rather have the release tag branching off one commit earlier, like this:
```
                         "5.8" (release)
                       /
                      /
"5.8.dev0" -- <stuff> -- "5.9.dev0" -- <stuff> (main)
```
That way, the main branch sees a steady progression of `*.dev0` versions, and each release branch also sees a progression from `X.dev0` to `X`, rather than having an extra `(X+1).dev0` appear (out of order) in between.

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (joshuacwnewton) [reply .github/workflows/create-release.yml:149]: Good catch! (Both with the `git add` part and the above branch structure part.)

**Comment by mguaypaq** [comment]
@joshuacwnewton, how does this look? (I sadly can't just add you as a reviewer, since you're the PR author, but we can pretend.)

**Review by joshuacwnewton — COMMENTED** [review]
Thank you so much for commit 79517a6ffc300eb3a50dd0efecb1b54cadef6de8. This looks much more sensible now. 

LGTM!

**Pull request merged**